prow/external-plugins/cherrypicker/server.go

```diff
@@ -105,19 +104,6 @@ type Server struct {
 	repos    []github.Repo
 }
 
-// ServeHTTP validates an incoming webhook and puts it into the event channel.
-func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	eventType, eventGUID, payload, ok, _ := github.ValidateWebhook(w, r, s.tokenGenerator)
-	if !ok {
-		return
-	}
-	fmt.Fprint(w, "Event received. Have a nice day.")
-
-	if err := s.handleEvent(eventType, eventGUID, payload); err != nil {
-		logrus.WithError(err).Error("Error parsing event.")
-	}
-}
-
 func (s *Server) handleEvent(eventType, eventGUID string, payload []byte) error {
```

Comment:
I think all of this functionality should be shared, too

Reply:
This func is different in each plugin. That's why we can pass this func in the handler.

Reply:
Meaning that if we force the plugins to manage only the `issue_comment` events.

Reply:
Maybe we could have an interface with different `HandleEventX` methods, and plugins that do not care about certain events would have nop implementations of the methods they don't care about?

Reply:
Handling issues and pull requests should be enough. I don't like the idea of creating an interface and forcing the plugins to implement empty functions if they don't need them. A nil check in the handler should be sufficient.